### DISCUSSION TITLE
Acronym mod select hotkey style

### DISCUSSION THREAD

**Discussion by verarr** [ppy/osu#26293] [Ideas]
Typing the acronym in mod select toggles the corresponding mod.

An example:

1. no mod is selected
2. user types `h`﻿`d` -> Hidden is selected
3. user types `d`﻿`t` -> Double Time is selected (so we have HD+DT)
4. user types `h`﻿`d` (again) -> Hidden is deselected (leaving us with only Double Time)
5. user types `n`﻿`c` -> Nightcore is selected, deselecting Double Time because they're incompatible
6. user types a backspace -> all mods are deselected

If the user types a backspace in the middle of a combination, it should cancel the combination instead of deselecting all mods.

In case an invalid combination is entered (for example `a`﻿`g` isn't a mod, or the single letter `z`, because no mod starts with z), there should be some visual or auditory indication that the user's input was not accepted.

Also, while only half of a combination has been entered, you could filter the mods shown to the ones whose abbreviation starts with the letter typed, to help the user type the other letter in the combination.

I do realize that now, searching for an acronym in the search bar and pressing enter pretty much achieves the same thing, but it is considerably slower to select multiple mods this way.

**Comment by Rekunan** [comment]
The "official" term is acronym.

If possible, I'd also like to see if multi selection for typing in multiple entries (ex: `DTEZHD` selects double time, easy, and hidden) is possible. Tho error handling def sounds way more of a hassle with that. Maybe auto select and then pop a warning when something invalid is typed in (ex: `DTZ` would select DT and then pop when `Z` is entered in)?
